Rework IBM Cloud Public instructions to call common instructions

ericvn opened the pull request.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/3662#" title="Author self-approved">ericvn</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **frankbu**

If they are not already assigned, you can assign the PR to them by writing `/assign @frankbu` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["frankbu"]} -->

ericvn commented:
/assign @frankbu

ericvn commented:
@linsun  FYI, I did slack with Long and got back: 
```Yes, seems that the steps in `public` piece is same to the common istio installation steps. I’m OK with that.

For `private` piece, we need to update the step once we have `ibm-istio` chart published to `catalog`.
```

frankbu commented:
If the generic instructions are all your pointing at, then I would not even have this section. Instead, just add a sentence at the top that says for IBM Cloud Public, just follow the Istio instructions and point to the Kubernetes getting started index.

That said, I think the IBM platform setup section there needs to be updated to remove the free cluster stuff, and add the necessary warnings and such that you just did in your previous PR.

ericvn commented:
@frankbu I further simplified the `IBM Cloud Public` section to a single line. I did leave the section since we have links for the 3 ways to use with IBM Cloud in the first paragraph.

A rebase should have fixed up the IKS free changes I did in the other PR.

One other note is that in the simplification, I removed a tip:

```
 {{< tip >}}
	    The Istio `demo` profile (`install/kubernetes/helm/istio/values-istio-demo.yaml`) is specified in the following command to support smaller machine-typed IBM Cloud Kubernetes Service clusters. If using a cluster of sufficient size, you can remove the `--values` parameter which will use the default Istio configuration values instead. See Istio's built-in [configuration profiles](/docs/setup/kubernetes/additional-setup/config-profiles/) for a description of the available profiles.
	    {{< /tip >}}
```

I think this is OK as the wording applies to all platforms and it is noted on the Profiles page that the **demo** profile is for modest resources. I added the _tip_ when we needed to use the **demo** profile to make things work on _free_ clusters and they are no longer supported.

frankbu commented on content/docs/setup/kubernetes/install/platform/ibm/index.md:
```diff
@@ -22,131 +22,7 @@ To install the managed Istio add-on in IBM Cloud Public, see the [IBM Cloud Kube
-    {{< text bash >}}
-    $ for i in install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl delete -f $i; done
-    {{< /text >}}
+To use Helm with IBM Cloud Public, follow the instructions for [install/uninstall](/docs/setup/kubernetes/install/helm/) and [upgrading](/docs/setup/kubernetes/upgrade).
```
```suggestion
Follow [these instructions](/docs/setup/kubernetes/install/helm/) to install and run the current release version of Istio in [IBM Cloud Public](https://www.ibm.com/cloud/) using Helm and the IBM Cloud Kubernetes Service (IKS).

To upgrade Istio in an existing IKS cluster, follow the [upgrade instructions](/docs/setup/kubernetes/upgrade) instead.
```

istio-testing commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/3662#" title="Author self-approved">ericvn</a>*, *<a href="https://github.com/istio/istio.io/pull/3662#pullrequestreview-215057916" title="Approved">frankbu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[content/OWNERS](https://github.com/istio/istio.io/blob/master/content/OWNERS)~~ [frankbu]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
